src/UI/Effect.elm

```diff
@@ -0,0 +1,58 @@
+module UI.Effect exposing
+    ( Effect
+    , SideEffect(..)
```

Comment:
not sure if this is the best time, but the need for a common package for effects is here. The question for me is, should this package have the definition of the `SideEffect` or should it be generic?

Reply:
- This PR is using the `Effect` type as an `OutMsg`.

It doesn't feel wrong but has some duplicated code as a result.
The alternatives are:

- Something like `Translator` the pattern (generic effects)
- Centralizing every `Effect` in a package, kinda following the `NoMap` pattern

The `Mixpanel` integration in `wms-web` is currently using the `Translator` pattern, but the API there is way simpler.